### PR TITLE
Allow the ISO to be booted from grub directly

### PR DESCRIPTION
**Pull request by Mexit**
This PR adds the ability to run an ISO image directly from grub.
This can be done using a simple menu, e.g:  [link](https://github.com/Mexit/MultiOS-USB/blob/master/config/grub_loopback/loopback.cfg).
Loopback compatible ISO images on your USB, HDD, SSD drive are automatically detected and you can boot from the selected ISO image. There is no need to burn an ISO image to disk. This allows you to have many different versions on one disk.

### CODE REVIEW

**Comment by schaefi** [comment]
Hey, thanks much for your work :+1: Looking at it today

File: kiwi/bootloader/config/grub2.py [hunk 1]
```diff
@@ -439,6 +439,12 @@ def setup_install_boot_images(self, mbrid, lookup_path=None):
                 lookup_path=lookup_path, mbrid=mbrid
             )
 
+        log.info('--> Creating loopback config')
+        loopback_file = self.boot_dir + '/' + self.get_boot_path() + '/' + self.boot_directory_name + '/loopback.cfg'
```
Comment (schaefi) [inline kiwi/bootloader/config/grub2.py:443]: Can this be done by a join, e.g

```python
loopback_file = os.path.join(
    self.boot_dir, self.get_boot_path(), self.boot_directory_name, '/loopback.cfg'
)
```

File: kiwi/bootloader/template/grub2.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def __init__(self):
                 menuentry "If OK, run snapper rollback and reboot." { true; }
               }
             fi
+            if [ -n "$${iso_path}" ]; then
+              isoboot="iso-scan/filename=$${iso_path}"
```
Comment (schaefi) [inline kiwi/bootloader/template/grub2.py:40]: nipick on 4 space indentation :)

**Review by schaefi — COMMENTED** [review]
First review just on python stuff...

**Comment by schaefi** [comment]
Just to put me right and to get things in order. The concept here is as follows:

* For install and live ISO images loopback.cfg is created which sources the grub.cfg
* The loopback.cfg is read in via a custom grub menu from "somewhere" (e.g on stick)
   ```
   menuentry ... {
       iso_path=...
       export iso_path
       loopback
       configfile (loop)/.../loopback.cfg
       loopback ... loop
   }
   ```
* The exported iso_path variable from that custom menu entry causes the setting of the isoboot variable
   ```isoboot="iso-scan/filename=$${iso_path}"``` which makes use of the iso-scan feature
* The final boot a menu entry from the grub.cfg in the image contains iso-scan/... in the kernel cmdline and can
   boot ISO media that supports the iso-scan feature

Did I summarize this correctly ?

**Comment by schaefi** [comment]
If I understood it correctly I think the missing part is in the documentation. It would be imho good to have an at least high level overview about this feature and how it can be used. Thoughts ?

**Comment by Mexit** [comment]
Sorry for my mistakes, these are my first steps with Python.

I agree, you summarised it well.

You can manually create, for example, such a grub menu:
```
menuentry "Boot from openSUSE ISO" {
	iso_path="/path/to/openSUSE.iso"
	export iso_path
	loopback loop "$iso_path"
	root=(loop)
	configfile /boot/grub2/loopback.cfg
	loopback --delete loop
}
```
The final kernel cmdline additionally includes:
`iso-scan/filename=/path/to/openSUSE.iso`

**Comment by schaefi** [comment]
ok, great. From a coding perspective this is good now, thanks :+1: 

Before merge we need to update the documentation and give users some information how to make use of the feature. As you said it requires some pre-work with the grub menu entry to make use of the feature. Could you add this documentation in a chapter here:

```doc/source/working_with_images```

This can be simple and straight forward.

Could you do this ?

Thanks much

**Comment by Mexit** [comment]
> Could you do this ?

I will try.

Some infos about this feature: https://git.kernel.org/pub/scm/boot/dracut/dracut.git/log/?qt=grep&q=iso-scan%2Ffilename

**Review by schaefi — APPROVED** [review]
Thanks for the docs, this looks great now :+1:

**Pull request merged**